presto-main-base/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -655,6 +656,12 @@ public PlanOptimizers(
                         ImmutableSet.of(new SimplifyCountOverConstant(metadata.getFunctionAndTypeManager()))),
                 new LimitPushDown(), // Run LimitPushDown before WindowFilterPushDown
                 new WindowFilterPushDown(metadata), // This must run after PredicatePushDown and LimitPushDown so that it squashes any successive filter nodes and limits
+                new IterativeOptimizer(
+                        metadata,
+                        ruleStats,
+                        statsCalculator,
+                        estimatedExchangesCostCalculator,
+                        ImmutableSet.of(new MinMaxByToWindowFunction(metadata.getFunctionAndTypeManager()))),
```

Comment:
Maybe we want to add this optimization before the window pushdowns? Let's make sure the topn=1 pushdown happens for this rewritten plan as well

Reply:
No but the pushdown of topn=1 (PartialTopN) looks like it already happened?